### PR TITLE
feat(number)!: change float default params

### LINKED ISSUE #1597: Change `number.float` default value from `precision` to `max`
We want to change the default value of `number.float` to `{ min: 0.0, max: 1.0, precision: 0.01 }` (or at least by default two digits after comma precision) (like is right now)

The default parameter just number parameter should change the `max` value, not the `precision`

This should not affect legacy `datatype.float`!

Waiting on:
- #1122

### PR DESCRIPTION
**Pull request by Shinigami92**
closes #1597

### CODE REVIEW

File: test/number.spec.ts [hunk 1]
```diff
@@ -143,13 +143,14 @@ describe('number', () => {
 
     describe('float', () => {
       it('should return a random float with a default precision of 2 digits after floating point', () => {
-        const number = faker.number.float();
-        expect(number).toBe(Number(number.toFixed(2)));
+        const float = faker.number.float();
```
Comment (ST-DDT) [inline test/number.spec.ts]: In a lot of other test we use `actual` maybe use that here instead?

**Review by ST-DDT — DISMISSED** [review]

**Review by ejcheng — DISMISSED** [review]

File: src/modules/number/index.ts [hunk 2]
```diff
@@ -62,18 +62,18 @@ export class NumberModule {
   /**
    * Returns a single random floating-point number for a given precision or range and precision.
    *
-   * @param options Precision or options object. Defaults to `{}`.
-   * @param options.min Lower bound for generated number. Defaults to `0`.
-   * @param options.max Upper bound for generated number. Defaults to `99999`.
+   * @param options Upper bound or options object. Defaults to `1.0`.
```
Comment (xDivisionByZerox) [inline src/modules/number/index.ts]: The default value is wrong here. It still is `{}`.

```suggestion
   * @param options Upper bound or options object. Defaults to `{}`.
```

File: test/__snapshots__/number.spec.ts.snap [hunk 3]
```diff
@@ -52,13 +52,13 @@ exports[`number > 1211 > bigInt > with string value 1`] = `24n`;
 
 exports[`number > 1211 > float > with max 1`] = `64.07`;
```
Comment (xDivisionByZerox) [inline test/__snapshots__/number.spec.ts.snap:53]: This doesn't look right :/
A `float(1)` or `float({ max: 1 })` should not return `64.07`. Or am I reading this wrong?

Comment (xDivisionByZerox) [reply test/__snapshots__/number.spec.ts.snap:53]: I'm stupid. the `max 1` means it is the 1st iteration with `max`. 🤦

Comment (Shinigami92) [reply test/__snapshots__/number.spec.ts.snap:53]: I exactly confused my self as well with this ^^'
It's not 1 as input, but the index 1 snapshot of vitest

**Review by xDivisionByZerox — APPROVED** [review]

**Review by ST-DDT — APPROVED** [review]

**Review by ejcheng — APPROVED** [review]

**Pull request merged**